pkg/monitortests/authentication/nodefaultserviceaccountoperatorstests/monitortest.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type noDefaultServiceAccountChecker struct {
+	kubeClient kubernetes.Interface
+	cfgClient  *configv1.ConfigV1Client
+}
+
+func NewAnalyzer() monitortestframework.MonitorTest {
+	return &noDefaultServiceAccountChecker{}
+}
+
+// Cleanup implements monitortestframework.MonitorTest.
+func (n *noDefaultServiceAccountChecker) Cleanup(ctx context.Context) error {
+	return nil
+}
+
+// generateDefaultSAFailures generates a list of failures where the pod in a list of pods
+// violated the default service account check.
+func generateDefaultSAFailures(podList []corev1.Pod) []string {
+	failures := make([]string, 0)
+	for _, pod := range podList {
+		podName := pod.Name
+		switch {
+		case strings.HasPrefix(podName, "cluster-version-operator-"), strings.HasPrefix(podName, "downloads-"),
+			strings.HasPrefix(podName, "etcd-guard-"), strings.HasPrefix(podName, "ingress-canary-"),
+			strings.HasPrefix(podName, "kube-apiserver-guard-"), strings.HasPrefix(podName, "kube-controller-manager-guard-"),
+			strings.HasPrefix(podName, "openshift-kube-scheduler-guard-"), strings.HasPrefix(podName, "monitoring-plugin-"),
+			strings.HasPrefix(podName, "multus-"), strings.HasPrefix(podName, "networking-console-plugin-"),
+			strings.HasPrefix(podName, "network-check-target-"), strings.HasPrefix(podName, "verify-all-openshiftcertifiedoperators-"),
+			strings.HasPrefix(podName, "verify-all-openshiftcommunityoperators-"), strings.HasPrefix(podName, "verify-all-openshiftredhatmarketplace-"):
+			continue
+		}
+		podSA := pod.Spec.ServiceAccountName
+		fmt.Printf("Service account for pod %s in namespace %s is %s\n", pod.Name, pod.Namespace, pod.Spec.ServiceAccountName)
+		// if the service account name is not default, we can exit for that iteration
+		if podSA != "default" {
+			continue
+		}
+		// otherwise, we need to flag the failure
+		failures = append(failures, fmt.Sprintf("service account name %s is being used in pod %s in namespace %s", podSA, pod.Name, pod.Namespace))
+	}
+	return failures
+}
+
+// CollectData implements monitortestframework.MonitorTest.
+func (n *noDefaultServiceAccountChecker) CollectData(ctx context.Context, storageDir string, beginning time.Time, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	if n.cfgClient == nil || n.kubeClient == nil {
+		return nil, nil, nil
+	}
+
+	namespaces, err := n.kubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	junits := []*junitapi.JUnitTestCase{}
+	for _, ns := range namespaces.Items {
+		// skip openshift-marketplace namespace for now - no actual pod name to report on
+		if ns.Name == "openshift-marketplace" {
+			continue
+		}
```

Comment:
I see what you are going for here, but I'm not sure it buys us much by skipping the pod listing for this namespace.

I'd recommend just having a common place for all exception behavior and moving this logic there.